dotcom-rendering/src/components/marketing/header/HeaderWrapper.tsx

```diff
@@ -0,0 +1,186 @@
+	ctaUrl: string;
+	ctaText: string;
+}
+
+export interface HeaderRenderedContent {
+	heading: string;
+	subheading: string;
+	primaryCta: HeaderEnrichedCta | null;
+	secondaryCta: HeaderEnrichedCta | null;
+	benefits: string[] | null;
+}
+
+export interface HeaderRenderProps {
+	content: HeaderRenderedContent;
+	mobileContent?: HeaderRenderedContent;
+	onCtaClick?: () => void; // only used by sign in prompt header
+}
+
+export const headerWrapper = (
+	Header: ReactComponent<HeaderRenderProps>,
+): ReactComponent<HeaderProps> => {
+	const Wrapped: ReactComponent<HeaderProps> = ({
+		content,
+		mobileContent,
+		tracking,
+		countryCode,
+		submitComponentEvent,
+		numArticles,
+	}) => {
+		const buildEnrichedCta = (cta: Cta): HeaderEnrichedCta => {
+			if (isProfileUrl(cta.baseUrl)) {
+				return {
+					ctaUrl: addTrackingParamsToProfileUrl(
+						cta.baseUrl,
+						tracking,
+					),
+					ctaText: cta.text,
+				};
+			}
+			return {
+				ctaUrl: addRegionIdAndTrackingParamsToSupportUrl(
+					cta.baseUrl,
+					tracking,
+					numArticles,
+					countryCode,
+				),
+				ctaText: cta.text,
+			};
+		};
+
+		const primaryCta = content.primaryCta
+			? buildEnrichedCta(content.primaryCta)
+			: null;
+		const secondaryCta = content.secondaryCta
+			? buildEnrichedCta(content.secondaryCta)
+			: null;
+		const benefits = content.benefits ?? null;
+
+		const renderedContent: HeaderRenderedContent = {
+			heading: content.heading,
+			subheading: content.subheading,
+			primaryCta,
+			secondaryCta,
+			benefits,
+		};
+
+		const mobilePrimaryCta = mobileContent?.primaryCta
+			? buildEnrichedCta(mobileContent.primaryCta)
+			: primaryCta;
+
+		const mobileSecondaryCta = mobileContent?.secondaryCta
+			? buildEnrichedCta(mobileContent.secondaryCta)
+			: secondaryCta;
+
+		const renderedMobileContent = mobileContent
+			? ({
+					heading: mobileContent.heading,
+					subheading: mobileContent.subheading,
+					primaryCta: mobilePrimaryCta,
+					secondaryCta: mobileSecondaryCta,
+			  } as HeaderRenderedContent)
+			: undefined;
+
+		const { abTestName, abTestVariant, componentType, campaignCode } =
+			tracking;
+
+		const onCtaClick = (componentId: string) => {
+			return (): void => {
+				const componentClickEvent = createClickEventFromTracking(
+					tracking,
+					`${componentId} : cta`,
+				);
+				if (submitComponentEvent) {
+					submitComponentEvent(componentClickEvent);
+				}
+			};
+		};
+
+		const sendOphanEvent = useCallback(
+			(action: OphanAction): void => {
+				if (submitComponentEvent) {
+					submitComponentEvent({
+						component: {
+							componentType,
+							id: campaignCode,
+							campaignCode,
+						},
+						action,
+						abTest: {
+							name: abTestName,
+							variant: abTestVariant,
+						},
+					});
+				}
+			},
+			[
+				abTestName,
+				abTestVariant,
+				campaignCode,
+				componentType,
+				submitComponentEvent,
+			],
+		);
+
+		const [hasBeenSeen, setNode] = useHasBeenSeen(
```

Comment:
Instead of migrating this hook over to DCR you can use the existing `useIsInView`.
This is [already the case for the epic](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/components/marketing/epics/ContributionsEpic.tsx#L321)